File: 0350. Intersection of Two Arrays II/Solution0350.cpp
#include<stdio.h>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        vector<int> res;
        int n1 = nums1.size(), i1 = 0;        
        int n2 = nums2.size(), i2 = 0;
        sort(nums1.begin(), nums1.end());        
        sort(nums2.begin(), nums2.end());
        while (i1 < n1 && i2 < n2) {
            if (nums1[i1] < nums2[i2]) {
                i1++;
            } else if (nums1[i1] > nums2[i2]) {
                i2++;
            } else {
                res.push_back(nums1[i1]);
                i1++;
                i2++;
            }
        }
        return res;
    }
};

File: 0350. Intersection of Two Arrays II/Solution0350.py
class Solution0350:
    def intersect(self, nums1: List[int], nums2: List[int]) -> List[int]:
        n1, n2, res = sorted(nums1), sorted(nums2), []
        p1 = p2 = 0
        while p1 < len(n1) and p2 < len(n2):
            if n1[p1] < n2[p2]:
                p1 += 1
            elif n2[p2] < n1[p1]:
                p2 += 1
            else:
                res.append(n1[p1])
                p1 += 1
                p2 += 1
        return res